src/cincinnati/mod.rs

```diff
@@ -130,16 +143,25 @@ fn find_update(
     );
 
     // Find booted deployment in graph.
-    let cur_position = match graph
+    let (cur_position, cur_release) = match graph
         .nodes
         .iter()
-        .position(|n| is_same_checksum(n, &booted_depl.checksum))
+        .enumerate()
+        .find(|(_, node)| is_same_checksum(node, &booted_depl.checksum))
     {
-        Some(pos) => pos,
+        Some(current) => current,
         None => return Ok(None),
     };
     drop(booted_depl);
 
+    // Evaluate and record whether booted OS is a dead-end release.
+    if let Some(reason) = evaluate_deadend(&cur_release) {
+        log::warn!(
```

Comment:
Will we emit this warning on each tick? I wonder here if it'd be cleaner to just warn once instead.

Reply:
Yes, I had the same doubt. The actual behavior we want is a bit more complex and consists of:
 1) a flip-flop logic which triggers only at the edge of state changes
 2) one (or more) well-known place that sysadmin should setup some automated monitoring on